bottlerocket-settings-models/modeled-types/src/shared.rs

```diff
@@ -1181,6 +1181,47 @@ mod test_positive_integer {
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// Input value that needs to be a non-zero value, but should not be greater
+/// than an i32::MAX.
+#[derive(Clone, Debug, PartialEq, Scalar)]
+pub struct NonZeroInteger {
+    pub inner: i32,
+}
```

Comment:
You are also disallowing negative values. Maybe the type should be named `PositiveInteger`?

Reply:
renamed.